File: String/anagram.java
// { Driver Code Starts
import java.util.*;
import java.lang.*;
import java.io.*;

class GFG {
    
	public static void main (String[] args)throws IOException {
	
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int t = Integer.parseInt(br.readLine());
		while(t-->0)
		{
		    String str = br.readLine();
		    String s1 = str.split(" ")[0];
		    String s2 = str.split(" ")[1];
		    
		    Solution obj = new Solution();
		    
		    if(obj.isAnagram(s1,s2))
		    {
		        System.out.println("YES");
		    }
		    else
		    {
		         System.out.println("NO");
		    }
		    
		    
		    
		}
	}
}// } Driver Code Ends

class Solution
{    
    //Function is to check whether two strings are anagram of each other or not.
    public static boolean isAnagram(String a,String b)
    {
        HashMap<Character,Integer> map=new HashMap<Character,Integer>();
        HashMap<Character,Integer> map1=new HashMap<Character,Integer>();
        // Your code here
        char[] aa=a.toCharArray();
        char[] bb=b.toCharArray();
        for (char x : aa){
            if(map.containsKey(x))
                map.put(x,map.get(x)+1);
            else
              map.put(x,1);
        }
        
        for (char x : bb){
            if(map1.containsKey(x))
                map1.put(x,map1.get(x)+1);
            else
              map1.put(x,1);
        }

        if(map.equals(map1))
        return true;
        else
        return false;
    }
}

File: String/anagram.py


#User function Template for python3


class Solution:
    
    #Function is to check whether two strings are anagram of each other or not.
    def isAnagram(self,a,b):
        #code here
        aa=list(a)
        bb=list(b)
        dict1={}
        dict2={}
        for x in aa :
            if x in dict1:
                dict1[x]=dict1[x]+1
            else:
                dict1[x]=1
                
        for x in bb :
            if x in dict2:
                dict2[x]=dict2[x]+1
            else:
                dict2[x]=1
                
        if dict1==dict2:
            return 1
        else:
            return 0
        
                

#{ 
#  Driver Code Starts
#Initial Template for Python 3

import atexit
import io
import sys

_INPUT_LINES = sys.stdin.read().splitlines()
input = iter(_INPUT_LINES).__next__
_OUTPUT_BUFFER = io.StringIO()
sys.stdout = _OUTPUT_BUFFER

@atexit.register

def write():
    sys.__stdout__.write(_OUTPUT_BUFFER.getvalue())

if __name__=='__main__':
    t = int(input())
    for i in range(t):
        a,b=map(str,input().strip().split())
        if(Solution().isAnagram(a,b)):
            print("YES")
        else:
            print("NO") 
# } Driver Code Ends
